x-pack/test/security_solution_endpoint/apps/endpoint/endpoint_transform.ts

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Client } from '@elastic/elasticsearch';
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../configs/ftr_provider_context';
+import { targetTags } from '../../target_tags';
+
+export default ({ getPageObjects, getService }: FtrProviderContext) => {
+  const retry = getService('retry');
+  const endpointTestResources = getService('endpointTestResources');
+  const esClient: Client = getService('es');
+  const sleep = (ms: number) => new Promise((f) => setTimeout(f, ms));
+
+  const transformAggregation = {
```

Comment:
Can you convert this to a function so that we do not use mutable data between tests.

Reply:
not sure I understand this one. I'm following the format of `endpoint_list.ts` in the same dir, that has a const up here. the `const expectedData = [ ... ]`


You want `const transformAggregation = () => ({ init_script: ... etc ... })`?

Reply:
correct. We should not share state (especially mutable objects) between tests. 

The other option on this one, since its a simple object (no nested objects), is to use `Object.freeze()` on it so that it can't be mutated.

Reply:
no prob, bob